internal/cmp/loadbalancer_datasource.go

```diff
@@ -0,0 +1,48 @@
+// (C) Copyright 2021 Hewlett Packard Enterprise Development LP
+
+package cmp
+
+import (
+	"context"
+	"fmt"
+	"log"
+
+	"github.com/HewlettPackard/hpegl-vmaas-cmp-go-sdk/pkg/client"
+	"github.com/HewlettPackard/hpegl-vmaas-terraform-resources/internal/utils"
+	"github.com/tshihad/tftags"
+)
+
+type loadBalancerds struct {
+	lbClient *client.LoadBalancerAPIService
+}
+
+func newLoadBalancerDS(loadBalancerClient *client.LoadBalancerAPIService) *loadBalancerds {
+	return &loadBalancerds{lbClient: loadBalancerClient}
+}
+
+func (n *loadBalancerds) Read(ctx context.Context, d *utils.Data, meta interface{}) error {
+	setMeta(meta, n.lbClient.Client)
+	log.Printf("[DEBUG] Get Load balancer")
+	name := d.GetString("name")
+
+	// Pre check
+	if err := d.Error(); err != nil {
+		return err
+	}
+
+	lb, err := n.lbClient.GetLoadBalancers(ctx)
+	if err != nil {
+		return err
+	}
+
+	for i, n := range lb.GetNetworkLoadBalancerResp {
```

Comment:
I hope the loadblancer list response is not paginated. Otherwise, I hope the pagination is handled either in the library or in program logic

Reply:
its handled in GO SDK repo https://github.com/HewlettPackard/hpegl-vmaas-cmp-go-sdk/blob/254ea8100f702ee8640ccb920a9ddb77baa9e465/pkg/models/load_balancer.go#L101